server/task_executor_pitr_restore.go

```diff
@@ -109,6 +116,18 @@ func (exec *PITRRestoreTaskExecutor) doPITRRestore(ctx context.Context, task *ap
 	return nil
 }
 
+// getAndCreateBinlogDirectory returns the path of a instance binlog directory.
+// nolint
+func getAndCreateBinlogDirectory(dataDir string, instance *api.Instance) (string, error) {
```

Comment:
Since we are doing restore on the database level, and the `dataDir` is at the instance level (bigger than database level), we can save it as a field of `mysql.Restore` in the `New` method. Then we don't need to pass it around in the implementation.

Reply:
SG. Stand with you.